pkg/deployments/package.json

```diff
@@ -31,8 +31,10 @@
     "test": "hardhat test ./**/test/*.ts",
     "ci:prepare-config": "ts-node ci/prepare-config.ts"
   },
+  "dependencies": {
+    "@balancer-labs/balancer-js": "workspace:*"
+  },
```

Comment:
Why this change? The deployments package does not depend on balancer-js

Reply:
there are some deployment tasks that import a few types from `balancer-js`.

I only got the error in the first build and not always, saying it didn't find `@balancer-js`. So it seems to be a race condition.

I can try to reproduce it again to get the exact message if needed.

Reply:
This is the error message:

```
➤ YN0000: [@balancer-labs/v2-deployments]: ../../pvt/common/setupTests.ts(7,32): error TS2307: Cannot find module '@balancer-labs/balancer-js' or its corresponding type declarations.
```

Shouldn't `pvt/common` depend on `balancer-js` as well? (note that adding this dependency doesn't fix the issue by itself).

Reply:
One more note about this one:
- I understand that compilation fails without building `balancer-js` because of this dependency here: https://github.com/balancer-labs/balancer-v2-monorepo/blob/116f33b9000aa7e18b35aa28f1e32cae35eb9039/pkg/deployments/hardhat.config.ts#L5 which I think can be removed.
- Even if we can 'compile' `deployments` without `balancer-js`, fork tests won't run because there are some type imports that need it. Since the fork tests are outside `src`, that dependency will go unnoticed when running `build`. Therefore, I think it does make sense to add `balancer-js` as a dependency.
- (related) I've noticed that `v2-common` and `v2-helpers` also use `balancer-js`, but they don't declare it as a dependency. Those two packages also don't have a build task, but if you tried building them with `tsc` (just like `deployments` build task) they'd fail. So maybe they should depend on `balancer-js` as well?

Reply:
> fork tests won't run because there are some type imports that need it. (...) Therefore, I think it does make sense to add balancer-js as a dependency.

This is not what 'dependency' means in the context of npm packages, however. A 'dependency' package is one that must be installed in order to run (use) the package in question, and will therefore be automatically installed when installing the root package.

`deployments` has no dependencies: you don't need any other package in order to use it (to load the json files, etc.). If we do need a package in order to e.g. run tests, then it's a `devDependency`, which means it must be installed to do development (same as linters, test tools, compilers, etc) - but it's not a dependency of the package _as an end product_.

Reply:
(we don't want to force users to install `balancer-js` when installing `deployments`).

Reply:
In that case, shouldn't we remove the `tsc` step from the `build` script? Otherwise `yarn build` will just fail

Reply:
I've added a note referring to this, and removed the `--topological` flag.

Forcing to build the packages in sequence and adding the explicit dependencies is an overkill; the note should be enough to consider in first-time builds. In CI this doesn't seem to be an issue.